### DISCUSSION TITLE
Know how shared_secret and identity_secret generate the verification code needed to log in to steam

### DISCUSSION THREAD

**Discussion by Aurora15304** [dyc3/steamguard-cli#269] [Q&A]
If I know identity_secret and shared_secret, what should I do to generate steam login verification code

**Comment by dyc3** [comment]
This is the format of an maFile. You can replace the fields in it with your own information, and then use `steamguard import --files <path to file>` to import it.

```json
{
    "account_name": "hydrastar2",
    "steam_id": 76561199155706900,
    "serial_number": "5958223490807687449",
    "revocation_code": "R12345",
    "shared_secret": "E2jwbqbphb3ddAAgPaU+ZFqDZFA=",
    "token_gid": "2de62c507a1fb878",
    "identity_secret": "DaW/TQJeh1vzq4FHr0a57HgS6EA=",
    "uri": "otpauth://totp/Steam:hydrastar2?secret=CNUPA3VG5GC33XLUAAQD3JJ6MRNIGZCQ&issuer=Steam",
    "device_id": "android:eaf2f3a4-7828-4f03-9182-9a7f1ae44d22",
    "secret_1": "7DLxRiDJLirJdTmRiu+IIXZlQKA=",
    "tokens": null,
}
```

Alternatively, if you have an maFile from SDA, you can import it with `steamguard import --sda --files <path to file>`

If you're looking for a more technical explanation, [this is how 2fa codes are generated](https://github.com/dyc3/steamguard-cli/blob/26c6916e795d926511b597a8a431e811754453a3/steamguard/src/token.rs#L31) and [this is how the identity_secret is used for confirmations](https://github.com/dyc3/steamguard-cli/blob/26c6916e795d926511b597a8a431e811754453a3/steamguard/src/confirmation.rs#L400).
